src/frontend/lib/views/insights/widgets/insights_selection.dart

```diff
@@ -1,85 +1,75 @@
+import 'package:cached_network_image/cached_network_image.dart';
 import 'package:flutter/material.dart';
 import 'package:frontend/models/insight_model.dart';
+import 'package:frontend/models/insight_type_model.dart';
 import 'package:frontend/providers/insight_choices_provider.dart';
-import 'menu_item.dart';
+import 'package:frontend/providers/insight_types_provider.dart';
 import 'package:provider/provider.dart';
 
-class InsightMenuItems {
-
-  static List<InsightMenuItem> choices = [
-    const InsightMenuItem('Pests', Icons.bug_report, InsightType.pest),
-    const InsightMenuItem('Diseases', Icons.sick, InsightType.disease),
-    const InsightMenuItem('Nutrient Deficiencies', Icons.water_drop, InsightType.nutrient),
-  ];
-
-}
-
-
 /// A widget that displays a hidden drawer that can be opened by swiping from the left edge of the screen
 class InsightsSelection extends StatefulWidget {
-  
   const InsightsSelection({super.key});
 
-
   @override
   _InsightsSelectionState createState() => _InsightsSelectionState();
-
 }
 
-class _InsightsSelectionState extends State<InsightsSelection> { 
-  
+class _InsightsSelectionState extends State<InsightsSelection> {
   @override
   void initState() {
     super.initState();
   }
 
-    Widget buildMenuItems(InsightMenuItem item) => StatefulBuilder(builder: (BuildContext context, StateSetter setState) => CheckboxListTile(
-      activeColor: Colors.green,
-      dense: true,
-      secondary: Icon(item.icon, color: Colors.black),
-      title: Text(
-        item.title,
-        style: TextStyle(fontSize: 13, letterSpacing: 0.5),
-      ),
-      value: Provider.of<InsightChoicesProvider>(context).selectedInsights.contains(item.type),
-      onChanged: (bool? isChecked) => {
-          setState(() {
-            Provider.of<InsightChoicesProvider>(context, listen: false).itemChange(item, isChecked!);
-          })
-      } ,
-    ));
-    
+  Widget buildMenuItem(InsightTypeModel item) => StatefulBuilder(
+      builder: (BuildContext context, StateSetter setState) => CheckboxListTile(
+            activeColor: Colors.green,
+            dense: true,
+            secondary: CachedNetworkImage(
+              imageUrl: item.icon,
+              height: 20,
+              width: 20,
```

Comment:
```suggestion
              width: 40,
```
This makes it a bit clearer maybe? But all in all, nice idea to make the icons align with the markers on the map :P
![image](https://user-images.githubusercontent.com/74115586/224508031-275da869-c961-4cfb-ad3c-a225e305553d.png)